demos/smart_classroom_demo/cpp/src/cnn.cpp

```diff
@@ -13,68 +13,74 @@
-    in.begin()->second->setLayout(InferenceEngine::Layout::NCHW);
-    input_blob_name_ = in.begin()->first;
-
-    InferenceEngine::OutputsDataMap out = cnnNetwork.getOutputsInfo();
-    for (auto&& item : out) {
-        item.second->setPrecision(InferenceEngine::Precision::FP32);
-        output_blobs_names_.push_back(item.first);
+
+    ov::preprocess::PrePostProcessor proc(cnnNetwork);
+    proc.input().tensor().
+      set_element_type(ov::element::f32).
+      set_layout({"NCHW"});
+    input_blob_name_ = cnnNetwork->input().get_any_name();
+    ov::OutputVector outputs = cnnNetwork->outputs();
+    for (auto&& item : outputs) {
+      proc.output(*item.get_names().begin()).tensor().set_element_type(ov::element::f32);
+      output_blobs_names_.push_back(item.get_any_name());
     }
+    cnnNetwork = proc.build();
 
     try {
-        executable_network_ = config_.ie.LoadNetwork(cnnNetwork, config_.deviceName);
-    } catch (const InferenceEngine::Exception&) {  // in case the model does not work with dynamic batch
-        cnnNetwork.setBatchSize(1);
-        executable_network_ = config_.ie.LoadNetwork(cnnNetwork, config_.deviceName,
-            {{InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_ENABLED, InferenceEngine::PluginConfigParams::NO}});
+        compiled_model_ = config_.ie.compile_model(cnnNetwork, config_.deviceName);
+    } catch (const ov::Exception&) {
+        ov::set_batch(cnnNetwork, 1);
+        compiled_model_ = config_.ie.compile_model(cnnNetwork, config_.deviceName);
     }
-    logExecNetworkInfo(executable_network_, config_.path_to_model, config_.deviceName, config_.model_type);
-    infer_request_ = executable_network_.CreateInferRequest();
+    logExecNetworkInfo(compiled_model_, config_.path_to_model, config_.deviceName, config_.model_type);
+    infer_request_ = compiled_model_.create_infer_request();
 }
 
 void CnnDLSDKBase::InferBatch(
         const std::vector<cv::Mat>& frames,
-        const std::function<void(const InferenceEngine::BlobMap&, size_t)>& fetch_results) const {
-    InferenceEngine::Blob::Ptr input = infer_request_.GetBlob(input_blob_name_);
-    const size_t batch_size = input->getTensorDesc().getDims()[0];
+        const std::function<void(const std::map<std::string, ov::runtime::Tensor>&, size_t)>& fetch_results) const {
+    ov::runtime::Tensor input = infer_request_.get_tensor(input_blob_name_);
+    const size_t batch_size = input.get_shape()[0];
 
     size_t num_imgs = frames.size();
     for (size_t batch_i = 0; batch_i < num_imgs; batch_i += batch_size) {
         const size_t current_batch_size = std::min(batch_size, num_imgs - batch_i);
         for (size_t b = 0; b < current_batch_size; b++) {
-            matToBlob(frames[batch_i + b], input, b);
+            matToTensor(frames[batch_i + b], input, b);
         }
 
-        if (config_.max_batch_size != 1)
-            infer_request_.SetBatch(current_batch_size);
-        infer_request_.Infer();
```

Comment:
Now this will always infer `batch_size` images, while it may need only `num_imgs - batch_i`

Reply:
U shouldn't just say "don't know". Describe what u tried and why it didn't work

Reply:
@Wovchena I think this was resolved by may patch

Reply:
You need to shrink `input_tensor` if `num_imgs < m_config.m_max_batch_size`

Reply:
yes, that's right, corrected.

cc @ivikhrev